### PR TITLE
Add tutorial for AutoScale launch_stack launchConfiguration

### PR DESCRIPTION
**Pull request by jasondunsmore**

### CODE REVIEW

**Comment by raj-patel** [comment]
To make this a fully working template, Can we add a resource that gets the web hook url for the autoscale group and add that to the output parameters section? In the curl, specify how to get the scale up url (output parameter).

File: autoscale-stack-launch-type.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        The image must have the software config agents baked-in.
+      default: f4bbbce2-50b0-4b07-bf09-96c175a45f4b
+
+It is important that the image being used has the the base OpenStack
+agents (``os-collect-config``, ``os-apply-config``, and
+``os-refresh-config``) baked into the image.
+
+Next, add a ``Rackspace::AutoScale::Group`` resource with a
+ ``launchConfiguration`` type of ``launch_stack``:
+
```
Comment (rs-randallburt) [inline autoscale-stack-launch-type.rst:77]: Missing code directive.

File: autoscale-stack-launch-type.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+To see the AutoScale group scale, create a "Scale Up" policy in the
+MyRackspace Portal along with an associated "Scale Up" webhook, and
+then trigger the webhook using curl:
+
+.. code:: bash
+
+    curl -X POST -I -k <webhook_scale_up_address_from_portal>
+
+You should now have two LAMP stacks and associated LAMP servers.
```
Comment (rs-randallburt) [inline autoscale-stack-launch-type.rst:199]: Needs a reference link to the full template.

Comment (jasondunsmore) [inline autoscale-stack-launch-type.rst:199]: Done

File: autoscale-stack-launch-type.rst (hunk 1 above)
Comment (jasondunsmore) [inline autoscale-stack-launch-type.rst:77]: Done

**Comment by rs-randallburt** [comment]
Mostly LGTM. My only concern is that the language sometimes seems to indicate that we're scaling a stack rather than using stacks instead of servers as our unit of scale. Not sure how to word it more clearly in the documentation, but might be something to think about for a future patch.

**Comment by jasondunsmore** [comment]
Randall, I reworded a sentence in the opening paragraph to make this more clear.

File: autoscale-stack-launch-type.rst [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+Brief summary
+=============
+
+In the *Event and schedule-based auto-scaling with Cloud
+Orchestration* tutorial, we used the "server" launchConfiguration type
+to scale the Cloud Servers in your web application.  Rackspace Auto
+Scale also supports a "stack" launchConfiguration type, where the unit
+of scale is a stack instead of a server.
+
+In this tutorial, we will learn how to scale a Heat stack, which will
```
Comment (rs-randallburt) [inline autoscale-stack-launch-type.rst]: much clearer, though I would change this a bit to be:

  In this tutorial, we will learn how to scale using a group of Heat stacks which allows us to scale configurations more complex than a single server. In this example, we will use a stack containing both a server and a `OS::Heat::SoftwareConfig` to configure new instances in the group.

**Comment by rs-randallburt** [comment]
Looking good. One more thing I was just reminded of: you'll need to add the new doc to the toc files as well or they won't show up in the index.

File: _toc.rst [hunk 4]
```diff
@@ -13,6 +13,7 @@
    cloud-backup
    cloud-database
    autoscale-heat-event-schedule-based
+   autoscale-stack-launch-type
```
Comment (rs-randallburt) [inline _toc.rst:16]: I think you have to add this to https://raw.githubusercontent.com/rackerlabs/rs-heat-docs/master/index.rst as well.

**Pull request merged**